### PR TITLE
doc: link to test examples

### PR DESCRIPTION
**Pull request by emillon**
See #9734

### CODE REVIEW

File: doc/hacking.rst [hunk 1]
```diff
@@ -114,6 +114,14 @@ For integration tests, we use a system similar to `Cram tests
...
 
+.. seealso::
+
+   `actions_to_sh tests <https://github.com/ocaml/dune/blob/3.12.2/test/expect-tests/dune_engine/action_to_sh_tests.ml>`_
+     An example of expect-tests.
+
+   `mdx-stanza/locks.t <https://github.com/ocaml/dune/blob/3.12.2/test/blackbox-tests/test-cases/mdx-stanza/locks.t>`_
+     An example of cram test.
+
 When running dune inside tests, the ``INSIDE_DUNE`` environment variable is set.
```
Comment (christinerose) [inline doc/hacking.rst:125]:
```suggestion
When running Dune inside tests, the ``INSIDE_DUNE`` environment variable is set.
```

File: doc/hacking.rst [hunk 2]
```diff
@@ -114,7 +114,15 @@ For integration tests, we use a system similar to `Cram tests
...
 .. _ppx_expect:      https://github.com/janestreet/ppx_expect
 
-When running dune inside tests, the ``INSIDE_DUNE`` environment variable is set.
+.. seealso::
+
+   `actions_to_sh tests <https://github.com/ocaml/dune/blob/3.12.2/test/expect-tests/dune_engine/action_to_sh_tests.ml>`_
+     An example of expect-tests.
+
+   `mdx-stanza/locks.t <https://github.com/ocaml/dune/blob/3.12.2/test/blackbox-tests/test-cases/mdx-stanza/locks.t>`_
+     An example of cram test.
```
Comment (christinerose) [inline doc/hacking.rst:123]:
```suggestion
     An example of Cram test.
```

**Review by christinerose — APPROVED** [review]

**Comment by christinerose** [comment]
The only question I have isn't part of your changes. In li. 27 "In order to build itself, Dune uses a micro dune written as a single" --- should it be micro Dune, micro `dune`, micro-Dune, or micro-dune?

File: doc/hacking.rst [hunk 3]
```diff
@@ -396,7 +404,7 @@ For links, prefer references that use ``:doc:`` (link to a whole document) or
 ``:term:`` (link to a definition in the glossary) to ``:ref:``.
 
 Use the right lexers:
-- ``dune`` for dune and related files
+- ``dune`` for Dune and related files
```
Comment (emillon) [inline doc/hacking.rst:407]:
```suggestion
- ``dune`` for ``dune`` and related files
```

**Comment by emillon** [comment]
Thamks, I extracted the unrelated edits to #9754.

**Pull request merged**